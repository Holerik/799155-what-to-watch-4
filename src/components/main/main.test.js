// main.test.js
import React from 'react';
import renderer from 'react-test-renderer';
import Main from './main.jsx';

const promoMovie = {
  id: 10,
  title: `The Grand Budapest Hotel`,
  poster: `img/the-grand-budapest-hotel-poster.jpg`,
  altPoster: `The Grand Budapest Hotel poster`,
  background: `img/bg-the-grand-budapest-hotel.jpg`,
  altBackground: `The Grand Budapest Hotel`,
  genre: [`Adventure`, `Crime`, `Comedy`],
  year: 2014,
  duration: `1h 39min`,
  age: `16+`,
  src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
};

const filmsInfo = [
  {
    id: 1,
    title: `Joker`,
    poster: `img/joker.jpg`,
    altPoster: `Joker poster`,
    background: `img/joker-bg.jpg`,
    altBackground: `Gotham City`,
    genre: [`Thriller`, `Crime`, `Drama`],
    year: 2019,
    duration: `2h 2min`,
    age: `18+`,
    src: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: 2,
    title: `The Commuter`,
    poster: `img/commuter.jpg`,
    altPoster: `The Commuter poster`,
    background: `img/commuter-bg.jpg`,
    altBackground: `New York City`,
    genre: [`Thriller`, `Action`, `Mystery`],
    year: 2018,
    duration: `1h 45min`,
    age: `16+`,
    src: `https://media.w3.org/2010/05/sintel/trailer.mp4`,
    preview: `https://media.w3.org/2010/05/sintel/trailer.mp4`,
  },
  {
    id: 3,
    title: `Molly's Game`,
    poster: `img/mollys-game.jpg`,
    altPoster: `Molly game poster`,
    background: `img/mollys-game-bg.jpg`,
    altBackground: `New York City`,
    genre: [`Drama`, `Crime`, `Biography`],
    year: 2017,
    duration: `2h 20min`,
    age: `18+`,
    src: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
];

const onMovieTitleClickHandler = jest.fn();

describe(`Main tests`, () => {
  it(`Main should render the screen`, () => {
    const tree = renderer
    .create(
        <Main
          promoMovie={promoMovie}
          filmsInfo={filmsInfo}
          onMovieTitleClick={onMovieTitleClickHandler}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
