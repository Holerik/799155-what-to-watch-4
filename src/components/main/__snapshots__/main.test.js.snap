// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main tests Main should render the screen 1`] = `
Array [
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="The Grand Budapest Hotel"
        src="img/bg-the-grand-budapest-hotel.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <div
          className="user-block__avatar"
        >
          <img
            alt="User avatar"
            height="63"
            src="img/avatar.jpg"
            width="63"
          />
        </div>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        >
          <img
            alt="The Grand Budapest Hotel poster"
            height="327"
            src="img/the-grand-budapest-hotel-poster.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            The Grand Budapest Hotel
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Thriller
            </span>
            <span
              className="movie-card__year"
            >
              2019
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              className="btn btn--list movie-card__button"
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
        onClick={[Function]}
        onMouseOver={[Function]}
      >
         
        <li
          className="catalog__genres-item
                            catalog__genres-item--active"
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            All genres
          </a>
        </li>
        <li
          className="catalog__genres-item
                            "
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Thriller
          </a>
        </li>
        <li
          className="catalog__genres-item
                            "
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Crime
          </a>
        </li>
        <li
          className="catalog__genres-item
                            "
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Drama
          </a>
        </li>
        <li
          className="catalog__genres-item
                            "
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Action
          </a>
        </li>
        <li
          className="catalog__genres-item
                            "
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Mystery
          </a>
        </li>
        <li
          className="catalog__genres-item
                            "
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Biography
          </a>
        </li>
      </ul>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
        >
          <div
            className="small-movie-card__image"
            id="1"
            onClick={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <img
              alt="Joker poster"
              height="175"
              src="img/joker.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href=""
              onClick={[Function]}
            >
              Joker
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <div
            className="small-movie-card__image"
            id="2"
            onClick={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <img
              alt="The Commuter poster"
              height="175"
              src="img/commuter.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href=""
              onClick={[Function]}
            >
              The Commuter
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <div
            className="small-movie-card__image"
            id="3"
            onClick={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <img
              alt="Molly game poster"
              height="175"
              src="img/mollys-game.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href=""
              onClick={[Function]}
            >
              Molly's Game
            </a>
          </h3>
        </article>
      </div>
      <div
        className="catalog__more"
      >
        <button
          className="catalog__button"
          type="button"
        >
          Show more
        </button>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
